src/app/chat/chat.service.ts

```diff
@@ -128,29 +160,39 @@ export class ChatService {
    *  size:20
    * }
    */
-  getMessageList(prams: messageListPrams): Observable<any> {
-    console.log("Service", this.messageList);
-    return of(this.messageList);
+  getMessageList(data: messageListPrams): Observable<any> {
+    return this.request.get(api.getChatMessages, {
+      params: data
+    });
   }
 
   markMessagesAsSeen(prams: markAsSeenPrams): Observable<any> {
-    prams.action = 'mark_seen';
-    return of("maked");
+    let body = {
+      'team_id': prams.team_id,
+      'id': prams.id,
+      'action': 'mark_seen'
+    }
+    return this.request.post(api.markAsSeen, body, {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+    });
   }
 
   postNewMessage(data: newMessage): Observable<any> {
-    data.env = 'develop';
-    let returnData = {
-      id: 300,
-      sender_name: 'Chathumal',
-      is_sender: true,
-      team_name: 'dream team',
-      sent_time: '1.30 PM',
-      message: data.message,
-      team_id: data.team_id,
-      to: data.to
-    };
-    return of(returnData);
+    const body = new HttpParams()
+      .set('data[to]', data.to.toString())
+      .set('data[message]', data.message)
+      .set('data[team_id]',data.team_id.toString())
+      .set('env', 'develop');
```

Comment:
Why do we pass the `env` as `develop`?

Reply:
we need to set env by checking the URL if it is sandbox we send develop, if it live we pass another thing. reason we passing that we back end need that env to identify the pusher event. in here i miss that URL checking part. i will fix that.